test/filters/oidc/jwks_resolver_test.cc

```diff
@@ -155,6 +155,39 @@ TEST(JwksResolverTest, TestDynamicJwksResolverWithInvalidHttpStatus) {
   EXPECT_EQ(nullptr, resolver.jwks());
 }
 
+TEST(JwksResolverTest,
+     TestDynamicJwksResolver_RequestBy3SecInterval_UntilJwksConfigured) {
+  boost::asio::io_context io_context;
+  auto mock_http = std::make_shared<common::http::HttpMock>();
+
+  // Configured request interval as 10000 sec. This value is enough to guarantee
+  // that second request will not be invoked for 3 sec evloop run.
+  DynamicJwksResolverImpl resolver("istio.io", std::chrono::seconds(10000),
+                                   mock_http, io_context);
+
+  // Never initialized with invalid HTTP status.
```

Comment:
If I understand correctly, we want to make the initial setup always return failure to check jwks as nullptr.

```
// Initially make mock server always return invalid HTTP response as 503.
```